### PR TITLE
Homotopy explicit all kwargs

### PR DESCRIPTION
**Pull request by CompRhys**
This PR seeks to address: https://github.com/pytorch/botorch/issues/2579

`optimize_acqf_homotopy` is a fairly minimal wrapper around `optimize_acqf` but didn't have all the constraint functionality.

This PR copies over all of the arguments that we could in principal want to use up into `optimize_acqf_homotopy`. For the time being `final_options` has been kept. The apparent bug with fixed features not being passed to the final optimization has been fixed. 

a simple dict rather than `OptimizeAcqfInputs` dataclass is used to store the shared parameters. 

## Related PRs

The original approach in https://github.com/pytorch/botorch/pull/2580 made use of kwargs which was opposed due to being less explicit.

### CODE REVIEW

File: botorch/optim/optimize_homotopy.py
Comment (saitcakmak) [inline botorch/optim/optimize_homotopy.py:21]: Could you run `ufmt format .` on the PR? This will fail the linter.

**Review by saitcakmak — APPROVED** [review]
Thanks for the update! This is much more in-line with the rest of the `optimize_acqf` signatures.

File: botorch/optim/optimize_homotopy.py
Comment (CompRhys) [reply botorch/optim/optimize_homotopy.py:21]: Sure, Can I add a pre-commit yaml to have this be an auto hook?

File: test/optim/test_homotopy.py
Comment (saitcakmak) [reply test/optim/test_homotopy.py:180]: Weird that this is failing. I was gonna try comparing floats but looks like I don't have commit access to the PR

Comment (CompRhys) [reply test/optim/test_homotopy.py:180]: If I make this change it at least makes the failure explict:

```python
FAILED test/optim/test_homotopy.py::TestHomotopy::test_optimize_acqf_homotopy - AssertionError: 1.999999999999982 not greater than or equal to 2.0
```

Would you be okay if I change the test to >= 2.0 - 1e-6?

Comment (saitcakmak) [reply test/optim/test_homotopy.py:180]: That sounds reasonable to me

**Comment by CompRhys** [comment]
Can we hold off merging this as I signed the CLA for a previous commit in my personal time and someone at my new company wants the lawyer to check

**Comment by saitcakmak** [comment]
Sure, let us know when it is safe to merge.

**Comment by CompRhys** [comment]
we're good to go

**Pull request closed without merging**
